packages/reporters/dev-server/src/ServerReporter.js

```diff
@@ -1,11 +1,15 @@
 // @flow
 
+import type {HTTPServer} from './types.js.flow';
+export type {HTTPServer};
+
 import {Reporter} from '@parcel/plugin';
 import invariant from 'assert';
 import Server from './Server';
 import path from 'path';
 
 let servers: Map<number, Server> = new Map();
+export let devServer: HTTPServer | null = null;
```

Comment:
Not sure if this is a good idea, maybe we should merge the hmr and dev server into one reporter so we don't have to share state between them.

Also won't this make the servers map unusable? This allowed parcel to run multiple servers at once, mainly when using the api this is possible and shouldn't break.

Reply:
> Not sure if this is a good idea, maybe we should merge the hmr and dev server into one reporter so we don't have to share state between them.

I agree that merging them would make the most sense (and I guess also address your other concert?), I just didn't want to start with that big a refactor. 🙂 

> Also won't this make the servers map unusable? This allowed parcel to run multiple servers at once, mainly when using the api this is possible and shouldn't break.

Hm, interesting, can you point me to a documentation section / article / project using such a setup?

Reply:
I can't point to a setup that does this, it was just to prevent future issues if anyone would ever create more than one parcel instance using the api that starts up a devserver.

Originally I thought it would maybe get used in a setup where someone wants to run multiple applications for example in a monorepo while sharing the workers and cache for better performance

Reply:
Oh, OK, I think I understand the use case, and it would most probably not work in the current form. So should I fix it, or refactor the whole thing and merge the reporters?

Reply:
I think so, merging the reporters seems like the best solution to fix this imo

Reply:
Out of curiosity, was there a reason to not do this (HMR websocket using the dev server) from the get-go?

Reply:
HMR and Dev server had a different port on Parcel 1 so I kept it that way in Parcel 2 and therefore also created 2 reporters

Not entirely sure why Devon made it like this originally